src/main/java/edu/cornell/kfs/module/receiptProcessing/CsvBatchInputFileTypeBase.java

```diff
@@ -125,23 +133,24 @@ public Object parse(byte[] fileByteContent) throws ParseException {
     protected void validateCSVFileInput(final List<String> expectedHeaderList, InputStream fileContents) throws
             IOException {
         //use csv reader to parse the csv content
-        CSVReader csvReader = new CSVReader(new InputStreamReader(fileContents, StandardCharsets.UTF_8));
-        List<String> inputHeaderList = Arrays.asList(csvReader.readNext());
-
-        String errorMessage;
-
-        if (!CollectionUtils.isEqualCollection(expectedHeaderList, inputHeaderList)) {
-            errorMessage = "CSV Batch Input File contains incorrect number of headers";
-            //collection has same elements, now check the exact content orders by looking at the toString comparisons
-        } else if (!expectedHeaderList.equals(inputHeaderList)) {
-            errorMessage = "CSV Batch Input File headers are different";
-        } else {
-        		errorMessage = validateDetailRowsContainExpectedNumberOfFields(expectedHeaderList, csvReader);
-        }
+        try (CSVReader csvReader = new CSVReader(new InputStreamReader(fileContents, StandardCharsets.UTF_8))) {
```

Comment:
Just to confirm, will the enclosed InputStreamReader be auto-closed when the CSVReader closes? Or does the reader need to be defined on a line prior to the CSVReader so that it will auto-close accordingly by the try-with-resources block?

Reply:
Please see the latest commit. 

Question: are the final keywords advisable?

Reply:
In the case of local variables, I'm fine either way with including or excluding the "final" keyword. If your code needs to have a certain local variable remain unchanged, make sure that the code at least treats it as "effectively" final if it's not explicitly declared final.